### PR TITLE
Improve computation result pipelining.

### PR DESCRIPTION
**Pull request by pattyshack**
1. Do less work in CommitCollection / compute view snapshot in hasher goroutine.
2. Compute chunks, chunk data packs, and end state as part of result collection.

Interestingly, combining the committer and event hasher goroutines has no performance impact.  I'm not sure why they were split that way in the first place.

### CODE REVIEW

**Review by janezpodhostnik — APPROVED** [review]
Technically looks OK to me, but I probably didn't catch all the nuances, as I'm not deeply familiar with this part.

**Review by SaveTheRbtz — APPROVED** [review]

**Comment by pattyshack** [comment]
bors merge

**Comment by pattyshack** [comment]
bors retry

**Comment by pattyshack** [comment]
bors merge

**Review by zhangchiqing — APPROVED** [review]

File: engine/execution/computation/computer/computer.go [hunk 1]
```diff
@@ -255,6 +257,7 @@ func (e *blockComputer) getRootSpanAndCollections(
 	collections = append(
 		collections,
 		collectionItem{
+			blockId:         blockId,
```
Comment (pattyshack) [inline engine/execution/computation/computer/computer.go:260]: integration test picked up the fact that i forgot to initialized the field here.  I'll added some unittest to verify the chunk entries as well.

**Comment by pattyshack** [comment]
bors merge

**Pull request merged**